### PR TITLE
C version of router compressor with bit field

### PR DESCRIPTION
**Pull request by alan-stokes**
added a default ordered dict and made the executable finder explode earlier.

### CODE REVIEW

File: unittests/test_exec_finder.py [hunk 1]
```diff
@@ -39,7 +43,11 @@ def test_find_in_two_places(tmpdir):
     w2.remove()
     assert ef.get_executable_path("abc.aplx") == str(w1)
     w1.remove()
-    assert ef.get_executable_path("abc.aplx") is None
+    try:
+        ef.get_executable_path("abc.aplx")
+        return False
+    except Exception:
+        return True
```
Comment (dkfellows) [inline unittests/test_exec_finder.py]: This is _not_ how you assert that an exception happened!

File: unittests/test_exec_finder.py [hunk 2]
```diff
@@ -13,7 +13,11 @@ def test_create_and_config(tmpdir):
 
 def test_find_in_no_places():
     ef = ExecutableFinder([])
-    assert ef.get_executable_path("abc.aplx") is None
+    try:
+        ef.get_executable_path("abc.aplx")
+        return False
+    except Exception:
+        return True
```
Comment (dkfellows) [inline unittests/test_exec_finder.py]: This is not how you assert that an exception happened!

File: unittests/test_progress_bar.py [hunk 3]
```diff
@@ -28,19 +31,25 @@ def test_with_operation(pbclass):
 
 @pytest.mark.parametrize("pbclass", [ProgressBar, DummyProgressBar])
 def test_check_length_full(pbclass):
+    logger_utils.reset()
     p = pbclass(2, None)
-    with pytest.raises(Exception):
```
Comment (dkfellows) [inline unittests/test_progress_bar.py]: This is (was) how you assert that an exception is thrown.

**Comment by dkfellows** [comment]
(Default ordered dicts have gone in via separate PR, #81)

**Comment by alan-stokes** [comment]
fixed ready for someone to merge

File: unittests/test_exec_finder.py (hunk 1 above)
Comment (alan-stokes) [reply unittests/test_exec_finder.py]: fixed

File: unittests/test_exec_finder.py (hunk 2 above)
Comment (alan-stokes) [reply unittests/test_exec_finder.py]: fixed

File: unittests/test_progress_bar.py (hunk 3 above)
Comment (alan-stokes) [reply unittests/test_progress_bar.py]: reflected

**Pull request merged**